### ISSUE TITLE
ECDSA when the hash length is greater than the group's order 

### ISSUE THREAD

**Issue by afsym** [cslashm/ECPy#17]
The library does not support the case where, either for ECDSA sign or ECDSA verify, the hash length is greater than the group's order.

As an example, it can be (not the optimal way) implemented like this:
- In ecdsa.py#_do_sign, right after `msg = int.from_bytes(msg, 'big')`:
```
nbyteslen = (n.bit_length() + 7) // 8
mbyteslen = (msg.bit_length() + 7) // 8

if nbyteslen < mbyteslen:
    # msg is the 'n.bit_length()' leftmost bits of the received msg.
    msg = int(hex(msg)[2:2*nbyteslen + 2], 16)
```

- In ecdsa.py#verify, right after `h = int.from_bytes(msg,'big')`:
```
nbyteslen = (n.bit_length() + 7) // 8
hbyteslen = (h.bit_length() + 7) // 8
if nbyteslen < hbyteslen:
    # h is the 'n.bit_length()' leftmost bits of the received h value.
    h = int(hex(h)[2:2*nbyteslen + 2], 16)
```

**Comment by cslashm** [comment]
Thanks for reporting. You're right.
Feel free to make a PR, else I will work on it this week.

**Comment by afsym** [comment]
Should be fixed by https://github.com/cslashm/ECPy/pull/18.

**Comment by matbok** [comment]
> Should be fixed by #18.

No, There is a bug if hash has at least 8 zero bits leading: indeed, in the #18 fix, the 'if' checks whether msg bit length is greater than N bit length, but it should check if (8 * msg_byte_length) > N_bit_length

**Comment by cslashm** [comment]
hmmm I'll check against RFC and internal Ledger OS code

**Comment by cslashm** [comment]
should be fixed in 1.2.4 @matbok
